projects/step-frontend/src/lib/modules/_common/interceptors/http-error.interceptor.ts

```diff
@@ -11,18 +18,41 @@ export class HttpErrorInterceptor implements HttpInterceptor {
   private handleHttpError(error: HttpErrorResponse): Observable<any> {
     console.error('network error', error);
     if (error.status !== 401) {
-      if (error.error?.errorName && error.error?.errorMessage) {
-        this._snackBar.open(error.error.errorName + ': ' + error.error.errorMessage, 'dismiss');
+      if (error.error) {
+        this.showError(error.error);
       } else if (error.name && error.message) {
-        this._snackBar.open(error.name + ': ' + error.message, 'dismiss');
+        this.showError(error.name + ': ' + error.message);
       } else {
-        this._snackBar.open(error.error, 'dismiss');
+        this.showError('Unknown HTTP error');
       }
     }
     return throwError(() => error);
   }
 
+  private handleAsyncError(response: HttpEvent<any>): HttpEvent<any> {
+    if (response instanceof HttpResponse && response.body?.error) {
+      console.error(`Non HTTP Error: ${response.body.error}`);
+      this.showError(response.body?.error);
+    }
+    return response;
+  }
+
+  private showError(error: any) {
+    if (error?.errorName && error?.errorMessage) {
+      this._snackBar.open(error.errorName + ': ' + error.errorMessage, 'dismiss');
+    } else if (error?.errorName || error?.errorMessage) {
+      this._snackBar.open(error.errorName || error.errorMessage, 'dismiss');
+    } else {
+      this._snackBar.open(error, 'dismiss');
+    }
+  }
+
   intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
-    return next.handle(req).pipe(catchError((error) => this.handleHttpError(error)));
+    return next.handle(req).pipe(
+      tap((response: HttpEvent<any>) => {
+        return this.handleAsyncError(response);
```

Comment:
the ```tap``` operator is used for side effects, a value should not be returned from it; I am not sure if it even works, but it would be better to use do something like ```switchMap((response: HttpEvent<any>) => of(this.handleAsyncError(response))```

Reply:
Good point, I guess tap is ignoring results all together, so we can just remove the return.